### DISCUSSION TITLE
slow performance compared to 2.10.4

### DISCUSSION THREAD

**Discussion by samiodeh1337** [TanStack/virtual#331] [General]
Hi everyone, I have tried to switch from 2.10.4 to 3.0.0-beta.2
I have a very big grid table with many features, each cell represents also a component with states,
the speed of 2.10.4 was so good compared to 3.0.0-beta.2 which is very slow.
so what's the problem?
thanks

**Comment by wuarmin** [comment]
@samiodeh1337 did you solve the problem?

**Reply by samiodeh1337** [reply]
No I'm still using the old version, will try the new one on few days

**Comment by christiankaindl** [comment]
Hi @samiodeh1337 I have just opened PR #353 which may help with your performance issues. Feel free to try it out and let me know if it fixes your issues :)

**Edit**: the PR was merged and is available in the latest beta.14

**Reply by samiodeh1337** [reply]
I'll try it as soon as possible and update you, thanks!

**Reply by wuarmin** [reply]
@samiodeh1337 did you try?

**Reply by samiodeh1337** [reply]
@christiankaindl @wuarmin I have tried the latest version beta. 54, and it's performance is better than 2.10.4, but I can't say it's the best

**Reply by christiankaindl** [reply]
@samiodeh1337 what Kind of performance issues do you experience?
